refactor: PostListing and SearchResults

jdrueckert opened the pull request:
- align blog and module search form layouts
- rename `SearchForm` folder, file and module-specific function to `ModuleSearchForm`
- rename `SearchForm` in `BlogSearchForm.jsx` to `BlogSearchForm`

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+---
+layout: post
+title: "TeraSpotlight: Extrem View Distances with LOD Chunks"
```
```suggestion
title: "TeraSpotlight: Extreme View Distances with LOD Chunks"
```

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+description: >
+  The idea of extreme view distances with LOD chunks is nearly as old as this project, and now it's finally a dream come true.
+author: Skaldarnar
+header-img: "/img/terasaturday/9/banner.jpg"
```
is that intentionally in the terasaturday folder

skaldarnar replied:
forgot to push latest changes - it now uses it's own banner image from the correct folder.

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+## 🤩 Spotlight
+
+Thanks to _@4D Enthusiast_ a long-time dream has finally come true. 
+You can experience the worlds of Terasology with [**extrem view and render distances**](https://github.com/MovingBlocks/Terasology/pull/4452) now!
```
```suggestion
You can experience the worlds of Terasology with [**extreme view and render distances**](https://github.com/MovingBlocks/Terasology/pull/4452) now!
```

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+> As an example, with view distance Near (9 * 7 * 9), it would take 16 LODs to have an area the size of the earth visible, and 22 LODs for the moon to be directly visible in the sky (although a block-based moon couldn't actually move).
+
+So, the available levels of details should be sufficient 👍 
+In addition to scaling down the chunk resolution the visible details are also limited to _world facets_ that support the scaling.
```
```suggestion
In addition to scaling down the chunk resolution, the visible details are also limited to _world facets_ that support the scaling.
```

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+
+So, the available levels of details should be sufficient 👍 
+In addition to scaling down the chunk resolution the visible details are also limited to _world facets_ that support the scaling.
+This is currently only implemented for the basic terrain generation, but we are looking forward to extending this to features like trees.
```
Maybe mention `CoreSampleGameplay` here so people know what to start to check it out?

skaldarnar replied:
It also works for JS I think :thinking:

jdrueckert commented on _posts/2021-02-07-teraspotlight-lod-chunks.md:
```diff
@@ -0,0 +1,89 @@
+</div>
+
+> Another issue I had to fix was that the far clipping plane was sometimes too close. 
+> There's a tradeoof in rendering between the accuracy with which it can be determined which of two things is closer (and therefore should be visible in front) and the maximum distance at which you can see something. 
```
```suggestion
> There's a trade-off in rendering between the accuracy with which it can be determined which of two things is closer (and therefore should be visible in front) and the maximum distance at which you can see something. 
```

jdrueckert committed:
refactor: PostListing and SearchResult

- destructure postEdges before handing it over to PostListing
- use PostListing to display cards everywhere

jdrueckert committed:
chore: align blog and module search form layouts

jdrueckert committed:
chore: align BlogSearchForm and ModuleSearchForm naming

jdrueckert committed:
chore: fix prettier, eslint, and sonarcloud complaints

skaldarnar commented on src/components/ModuleSearchForm/ModuleSearchForm.jsx:
```diff
@@ -12,7 +12,7 @@ function SearchForm({ query, filter, prefix }) {
   return (
     <Form role="search" method="GET">
-      <Row form id="search-form">
+      <Row form className="justify-content-center" d="search-form">
```
```suggestion
      <Row form className="justify-content-center" id="search-form">
```

skaldarnar commented on src/components/SearchResult/SearchResult.jsx:
```diff
@@ -13,30 +11,13 @@ function SearchResults({ query, results }) {
-          )}
-        </Row>
-      )}
+      <PostListing prefix={`/${type}`} postList={results} />
```
Probably the `SearchResult` should receive the `prefix` in addition to the `type`, they don't need to be the same (they currently just happen to be the same)

jdrueckert committed:
chore: address review comments
